### DISCUSSION TITLE
How to build and test jekyll gem from source?

### DISCUSSION THREAD

**Discussion by velle** [jekyll/jekyll#9540] [Q&A]
Ruby newb here. I would appreciate your help :) Is there a tutorial on building, testing, installing from source? I could not find one. 

I managed to clone and build jekyll. 

    git clone git@github.com:jekyll/jekyll.git
    cd jekyll
    bundle install
    rake build

This builds `pkg/jekyll-4.3.3.gem`.

And now I would like to test, so I try:

    rake test
    
This prints a wall of text including a backtrace (the complete output is here: https://gist.github.com/velle/7f6e73ad1ab3ca830d9b8553f47fd939#file-gistfile1-txt), but the essence in the error is:

    cannot load such file -- simplecov (LoadError)

However, it seems like simplecov _has_ been installed. 

    $ find . -name "*simplecov*"
    ./test/simplecov_custom_profile.rb
    ./gems/ruby/3.0.0/cache/simplecov-0.22.0.gem
    ./gems/ruby/3.0.0/cache/simplecov-html-0.12.3.gem
    ./gems/ruby/3.0.0/cache/simplecov_json_formatter-0.1.4.gem
    ./gems/ruby/3.0.0/specifications/simplecov_json_formatter-0.1.4.gemspec
    ./gems/ruby/3.0.0/specifications/simplecov-html-0.12.3.gemspec
    ./gems/ruby/3.0.0/specifications/simplecov-0.22.0.gemspec
    ./gems/ruby/3.0.0/gems/simplecov_json_formatter-0.1.4
    ./gems/ruby/3.0.0/gems/simplecov_json_formatter-0.1.4/lib/simplecov_json_formatter.rb
    ./gems/ruby/3.0.0/gems/simplecov_json_formatter-0.1.4/lib/simplecov_json_formatter
    ./gems/ruby/3.0.0/gems/simplecov-0.22.0
    ./gems/ruby/3.0.0/gems/simplecov-0.22.0/lib/minitest/simplecov_plugin.rb
    ./gems/ruby/3.0.0/gems/simplecov-0.22.0/lib/simplecov
    ./gems/ruby/3.0.0/gems/simplecov-0.22.0/lib/simplecov.rb
    ./gems/ruby/3.0.0/gems/simplecov-html-0.12.3
    ./gems/ruby/3.0.0/gems/simplecov-html-0.12.3/lib/simplecov-html
    ./gems/ruby/3.0.0/gems/simplecov-html-0.12.3/lib/simplecov-html.rb
    ./gems/ruby/3.0.0/gems/simplecov-html-0.12.3/simplecov-html.gemspec
    ./gems/ruby/3.0.0/gems/cucumber-5.1.3/lib/simplecov_setup.rb

Am I doing it wrong?

**Comment by mattr-** [comment]
`bundle exec rake test` should be what you need to run

**Reply by velle** [reply]
Thanks :)
